src/components/tx-flow/flows/RecoveryAttempt/RecoveryAttemptReview.tsx

```diff
@@ -0,0 +1,93 @@
+import { type SyntheticEvent, useState, useContext, useCallback } from 'react'
+import { CircularProgress, CardActions, Button, Typography, Stack, Divider } from '@mui/material'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import CheckWallet from '@/components/common/CheckWallet'
+import { Errors, trackError } from '@/services/exceptions'
+import { dispatchRecoveryExecution } from '@/features/recovery/services/recovery-sender'
+import useWallet from '@/hooks/wallets/useWallet'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import type { TransactionAddedEvent } from '@gnosis.pm/zodiac/dist/cjs/types/Delay'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import TxCard from '@/components/tx-flow/common/TxCard'
+import FieldsGrid from '@/components/tx/FieldsGrid'
+import { TxModalContext } from '../..'
+
+export type RecoveryAttemptReviewProps = {
+  params: {
+    args: TransactionAddedEvent.Log['args']
+    address: string
+  }
+}
+
+const RecoveryAttemptReview = ({ params }: RecoveryAttemptReviewProps) => {
+  const [isPending, setIsPending] = useState(false)
+  const [error, setError] = useState<Error>()
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+  const { setTxFlow } = useContext(TxModalContext)
+
+  const onFormSubmit = useCallback(
+    async (e: SyntheticEvent) => {
+      e.preventDefault()
+
+      if (!wallet) return
+
+      setError(undefined)
+      setIsPending(true)
+
+      try {
+        await dispatchRecoveryExecution({
+          provider: wallet.provider,
+          chainId: safe.chainId,
+          args: params.args,
+          delayModifierAddress: params.address,
+          signerAddress: wallet.address,
+        })
+        setTxFlow(undefined)
+      } catch (err) {
+        trackError(Errors._812, err)
+        setError(err as Error)
+      }
+
+      setIsPending(false)
```

Comment:
It would be nice if we could move the logic of handling the `error` and `pending` states of the request to a separated hook, so we don't need to explicitly handle this cases every time we have to send a "dispatch request"

Reply:
Could you give me an example of how you envision this? I would use the useAsync hook here but it has to happen on form submit, and I'm not sure what would trigger the hook.

Reply:
similar to the way of how useQuery does, something like:

```jsx

function useAsyncDispatch<T, E>(dispatch: (props: T) => Promise<E>) {
  const [isLoading, toggleIsLoading] = useReducer((isLoading: boolean) => !isLoading, false)
  const [error, setError] = useState<Error>()

  const asyncDispatch = async (args: T) => {
    toggleIsLoading()
    setError(undefined)

    try {
      const data = await dispatch(args)

      toggleIsLoading()

      return data
    } catch (e) {
      setError(e as Error)
      toggleIsLoading()
    }
  }

  return { isLoading, error, asyncDispatch }
}
```
and in your component you could do:

```jsx
 const MyComponent = () => {
  const { asyncDispatch, isLoading, errror } = useAsyncDispatch<MyExpectedArgs, MyExpectedResult>();

 const onSubmit = async () => {
  const data = await asyncDispatch();

  // do somthing else
 }
 }
```

Reply:
I see, thanks!